File: Data Structures & Algorithms/string-encode-and-decode/submission-1.py
class Solution:

    def encode(self, strs: List[str]) -> str:
        encoded_string = ""
        for s in strs:
            encoded_string+= s+'0*0'
        return encoded_string

    def decode(self, s: str) -> List[str]:
        ans = s.split('0*0')[:-1];
        print(ans)
        return ans
        

File: Data Structures & Algorithms/anagram-groups/submission-1.cpp
class Solution {
   public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        vector<vector<string>> ans;
        unordered_map<string, vector<string>> mp;
        for (auto x : strs) {
            // cout<<x<<endl;
            string temp = x;
            sort(temp.begin(), temp.end());
            // cout<<temp<<endl;
            mp[temp].push_back(x);
        }

        for(auto x: mp){
            // for(auto y: x.second){
            //     cout<<y<<" ";
            // }
            ans.push_back(x.second);
        }

        return ans;
    }
};
